libs/wire-subsystems/src/Wire/InvitationCodeStore.hs

```diff
@@ -60,16 +64,23 @@ recordInstance ''StoredInvitationByTeam
 
 data InvitationCodeStore :: Effect where
   LookupInvitation :: TeamId -> InvitationId -> InvitationCodeStore m (Maybe StoredInvitation)
-  LookupInvitationCodesByEmail :: Email -> InvitationCodeStore m [StoredInvitationByTeam]
+  LookupInvitationInfo :: InvitationCode -> InvitationCodeStore m (Maybe (TeamId, InvitationId))
+  LookupInvitationCodesByEmail :: EmailAddress -> InvitationCodeStore m [StoredInvitationByTeam]
 
 makeSem ''InvitationCodeStore
 
-lookupInvitationByEmail :: (Member InvitationCodeStore r, Member TinyLog r) => Email -> Sem r (Maybe StoredInvitation)
-lookupInvitationByEmail email = runMaybeT do
-  MkStoredInvitationByTeam {teamId, invId} <- MaybeT $ lookupSingleInvitationCodeByEmail email
-  MaybeT $ lookupInvitation teamId invId
+-- TODO: account for show/hide?
+lookupInvitationByEmail :: (Member InvitationCodeStore r, Member TinyLog r) => ShowOrHideInvitationUrl -> EmailAddress -> Sem r (Maybe StoredInvitation)
```

Comment:
I changed this on purpose. ShowOrHideUrl is not part of the store. It should be a pure function outside of the store.